aten/src/ATen/templates/TypeExtensionDerived.h

```diff
@@ -0,0 +1,15 @@
+#pragma once
+#include <ATen/${Backend}Type.h>
+
+namespace at {
+
+struct CAFFE2_API ${Type} : public ${Backend}Type {
```

Comment:
Would be nice to have a doc saying what this class is used for

Reply:
Now that we are generating derived Types, should we still have a separate mechanism?  My concern here is:
1) The 'base' is called XLAType but it's really a TypeDefault which is strange.
2) There are still landmines, e.g. XLAFloatType::elementSizeInBytes throws an exception instead of returning the right answer.

Reply:
Actually, on further reflection I think Greg is right. But it would involve a differently involved rewrite to the code here.

Reply:
Agreed. Right now there's a bit too much coupling between backends and scalar types in ATen in general, and I plan to follow up on that in the future.